src/fetch/fees.py

```diff
@@ -0,0 +1,541 @@
+                executed_buy_amount,
+            ) = compute_volume_protocol_fee(
+                row,
+                executed_sell_amount,
+                executed_buy_amount,
+                volume_factor,
+            )
+
+        else:
+            raise ValueError(
+                f"Unknow protocol fee kind \"{row['kind']}\"."
+                'Only "surplus", "priceimprovement", and "volume" are supported.'
+            )
+
+        protocol_fees.append(protocol_fee)
+        protocol_fee_tokens.append(protocol_fee_token)
+        protocol_fee_token_native_prices.append(protocol_fee_token_native_price)
+        protocol_fee_kinds.append(protocol_fee_kind)
+
+    network_fee = compute_network_fee(row, executed_sell_amount, executed_buy_amount)
+
+    return (
+        int(row["auction_id"]),
+        str(row["order_uid"]),
+        str(row["winning_solver"]),
+        network_fee,
+        protocol_fees[::-1],
+        protocol_fee_tokens[::-1],
+        protocol_fee_token_native_prices[::-1],
+        protocol_fee_kinds[::-1],
+        is_partner_fee,
+        recipient,
+    )
+
+
+def compute_surplus_protocol_fee(
+    row: Series,
+    sell_amount: int,
+    buy_amount: int,
+    surplus_factor: float,
+    surplus_max_volume_factor: float,
+) -> tuple[int, str, float, int, int]:
+    """Compute the surplus protocol fee"""
+    sell_amount_before_fee = sell_amount
+    buy_amount_before_fee = buy_amount
+    if row["kind"] == "sell":
+        surplus = buy_amount - int(
+            sell_amount
+            * Fraction(int(row["limit_buy_amount"]), int(row["limit_sell_amount"]))
+        )
+        surplus_fee = int(surplus_factor / (1 - surplus_factor) * surplus)
+
+        volume = buy_amount
+        volume_fee = int(
+            surplus_max_volume_factor / (1 - surplus_max_volume_factor) * volume
+        )
+
+        protocol_fee = min(surplus_fee, volume_fee)
+        protocol_fee_token = row["buy_token"]
+        protocol_fee_token_native_price = row["buy_token_native_price"] / 10**18
+
+        buy_amount_before_fee += protocol_fee
+    elif row["kind"] == "buy":
+        volume = sell_amount
+        surplus = (
+            int(
+                buy_amount
+                * Fraction(int(row["limit_sell_amount"]), int(row["limit_buy_amount"]))
+            )
+            - sell_amount
+        )
+        surplus_fee = int(surplus_factor / (1 - surplus_factor) * surplus)
+
+        volume = sell_amount
+        volume_fee = int(
+            surplus_max_volume_factor / (1 + surplus_max_volume_factor) * volume
+        )
+
+        protocol_fee = min(surplus_fee, volume_fee)
+        protocol_fee_token = row["sell_token"]
+        protocol_fee_token_native_price = row["sell_token_native_price"] / 10**18
+
+        sell_amount_before_fee -= protocol_fee
+    else:
+        raise ValueError(
+            f"Unknow order kind \"{row['kind']}\". Only \"sell\" and \"buy\" are supported."
+        )
```

Comment:
Would be cool if pandas supported enums so you didn't have to put all these error cases everywhere. 

```suggestion
        raise ValueError(
            f"Unknown order kind \"{row['kind']}\". Only \"sell\" and \"buy\" are supported."
        )
```

Note that this `Unknow order kind` typo appears in multiple places that I won't continue to point out.